### PR TITLE
ports/stm32/i2c: Fix the code to support I2C4.

### PR DESCRIPTION
**Pull request by iabdalkader**
The current code assumes all I2Cs are on the same peripheral bus, which is not true for I2C4 and the same goes for the clock enable code.

Note I don't like this fix because it will not extend very well into the future (I see I2C5 in the HAL) so maybe it's best and more simple to just pass the I2C ID along with the handle and use the `__HAL_RCC_I2Cx_CLK_ENABLE()`  macros.

### CODE REVIEW

File: ports/stm32/i2c.c [hunk 1]
```diff
@@ -285,8 +285,22 @@ int i2c_write(i2c_t *i2c, const uint8_t *src, size_t len, size_t next_len) {
...
+static uint32_t i2c_get_id(i2c_t *i2c) {
+    // Mask the lower 16 bits to get the bus base address
+    uint32_t base = ((uint32_t)i2c) & 0xFFFF0000;
+
+    switch (base) {
+        #if defined(STM32H7)
+        case D3_APB1PERIPH_BASE:
+            return 3;
+        #endif
+        default:
```
Comment (dpgeorge) [inline ports/stm32/i2c.c:297]: I suggest making this simpler like this:
```c
#if defined(STM32H7)
if (i2c == I2C4) {
    return 3;
}
#endif

return ((uint32_t)i2c - I2C1_BASE) / (I2C2_BASE - I2C1_BASE);
```

Comment (iabdalkader) [reply ports/stm32/i2c.c:297]: Okay will change it to that, but the idea was if more I2C are added to D3, then they can be used to calculate the `i2c_id` same way it's done from D2.

Comment (dpgeorge) [reply ports/stm32/i2c.c]: It looks like the only I2C5 that's currently available is on D2_APB1PERIPH_BASE

Comment (iabdalkader) [reply ports/stm32/i2c.c]: I think I saw LPI2Cs in the drivers too, not sure which domain they belong to. I changed it to the above suggestion, using the base pointer, and did the same for the clock enable.

Comment (iabdalkader) [reply ports/stm32/i2c.c]: Actually I2C3 is not always defined, I reverted the clock enable to using IDs.

**Pull request merged**